prov/efa/src/efa_fabric.c

```diff
@@ -479,11 +491,39 @@ static int efa_get_gdr_support(char *device_name)
 		goto out;
 	}
 
-	ret = fi_read_file(get_sysfs_path(), gdr_path, &gdr_support, sizeof(char));
-	if (ret < 0)
+	ret = fi_read_file(get_sysfs_path(), gdr_path,
+			   gdr_support_str, MAX_GDR_SUPPORT_STRLEN);
+	if (ret < 0) {
+		if (errno == ENOENT) {
+			/* sysfs file does not exist, gdr is not supported */
+			ret = 0;
+		}
+
 		goto out;
+	}
+
+	if (ret == 0) {
+		EFA_WARN(FI_LOG_FABRIC, "Sysfs file %s is empty\n", gdr_path);
+		ret = -FI_EINVAL;
+		goto out;
+	}
+
+	gdr_support_str[MAX_GDR_SUPPORT_STRLEN-1] = '\0';
+	ret = strtol(gdr_support_str, &endptr, 10);
+	if ((endptr - gdr_support_str) != 1) {
```

Comment:
argh; this hurts my brain.  Why are you assuming that there's exactly 1 char?  Just use the regular check methodology and then check the integer for the types you want.  Stop trying to be cute.

Reply:
Not sure what is regular check here.

I changed it to 
```
if (endptr == gdr_support_str) {
    /* error handling */
} 
```
In new revision, in case the string does not contain digits.